qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteCodeActions.java

```diff
@@ -186,15 +199,79 @@ private static void doCodeActionToDisableValidation(Template template, List<Diag
 		codeActions.add(disableValidationForTemplateQuickFix);
 	}
 
+	/**
+	 * Create CodeAction for deprecated `with` Qute syntax.
+	 *
+	 * @param template    the Qute template.
+	 * @param diagnostic  the diagnostic list that this CodeAction will fix.
+	 * @param codeActions the list of CodeActions to perform.
+	 * @throws BadLocationException
+	 *
+	 */
+	private void doCodeActionsForNotRecommendedWithSection(Template template, Diagnostic diagnostic,
+			List<CodeAction> codeActions) {
+		Range withSectionRange = diagnostic.getRange();
+		try {
+			int withSectionStart = template.offsetAt(withSectionRange.getStart());
+			WithSection withSection = (WithSection) template.findNodeAt(withSectionStart + 1);
+			String javaResourceClassName = withSection.getObjectParameter().getName();

```

Comment:
```java

// 2. Collect all object parts from expression (ex: {name}) or section parameter (ex : {#let itemName=name})
```

Reply:
if your #with doesn't define parameter like item (ex : `{#with}`, you will have a NPE, in other words do:

```java
Parameter object = withSection.getObjectParameter();
if (object != null) {
	String javaResourceClassName = object.getName();
```

please write a test with this usecase and add test with code action and expected text edit.

Reply:
Please rename `javaResourceClassName` with `objectPartName`